### DISCUSSION TITLE
Welcome to dbus-serialbattery Discussions!

### DISCUSSION THREAD

**Discussion by Louisvdw** [Louisvdw/dbus-serialbattery#55] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Assist others.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

To get started, comment below with an introduction of yourself and tell us about what BMS/Battery you use. We also love to see pictures of your system if you have.

Please add multiline code or logfiles by useing three backticks (```) or three tildes (~~~) on the lines before and after the code block. See https://www.markdownguide.org/extended-syntax/#fenced-code-blocks for more details.

**Comment by SanderV4n** [comment]
Ill kick-off, Sander here! Electronics engineer by profession at a small IoT sensors company.

I contributed the initial Sinowealth code because I was planning on building a serial-to-can converter to integrate my battery/bms in the Venus OS. When I found this project I the microcontroller project went overboard, writing a bit of python was a lot less work.

My setup consists of:
12x 280Ah lishen cells in 3x4 configuration, giving 840Ah at 12V
Daly 250A smart bms with uart interface connected a Victron GX device
Victron Multiplus 300 as charger/inverter, charging at 100A

This is all installed on our boat, a Victoire 1122 (11 meters/37 feet sailing yacht)

For the discussion:
I was looking at the VRM advanced page with a card for the BMS:
![image](https://user-images.githubusercontent.com/23108348/135469578-2dc59937-e7d6-43f7-b989-f7dccb1c7579.png)
Here the used Ah is missing/not filled in, so I looked in the victron documentation for DBUS. It seems we are missing "/ConsumedAmphours", so this is a feature on my todolist for this project. (might take a while as the harbour internet connection to my boat is very flaky and the sailing season is winding down here in the Netherlands)

**Comment by Louisvdw** [comment]
I'm a software engineer and created the driver because my first set of cells were are mismatch of 80Ah cells (some only doing 74Ah) and that caused me enless problems when the ESS system did not know about the battery stats. Linking the BMS started that all out.

My setup consists of:
15x 120Ah LiFePo4 cells 48V with 100A LLT BMS
15x ~80Ah LiFePo4 cells 48V with 60A JKBMS
Daly 250A smart bms with uart interface connected a Victron GX device
Victron Multiplus II 3000  and 2x MPPT 100/20-48V

**Reply by smartpowernorway** [reply]
Hi Louis, 
Beforehand, I want to thank you for the great effort you have done. Your achievement helps a large number of people including myself.

I have a 12KWh LiFePO4 16 cells with a 150A Daly Smart BMS connected to my Victron power system: Multiplus 48/3K, Cerbo & MPPT.
Following you instructions, I managed to communicate the Daly to the Cerbo using the Daly's original USB-UART cable.
Now, while testing the unit, I induced some problems to the BMS (disconnect one cell, etc) and realized that the BMS remains ON meaning it does not disconnect the battery power.

Then tested removing the complete 17-pin connector of the BMS and also it remains ON. Looks like some voltage from the Cerbo is feeding back to the BMS on the UART port keeping the FETS switched ON.

Have you seen this issue? Did I do something incorrectly?

I am currently using Cerbo Venus OS 2.7. Will try today with 2.8

Thank you very much,

Ibrahin

**Reply by Louisvdw** [reply]
Hi Ibrahin
I suspect that you also connected the Red power wire that is part of your comms cable.
It is for this reason that I mention under 3) for Requirements that you should only connect the Ground, Rx & Tx to the BMS unless using a isolated cable like the Victron cables. 

Isolated cables means there is no cable link between the BMS side and the GX side, so no power can flow from one to the other and you need to supply power from both sides (the USB and the battery Red power wire). Most cables are not isolated, so the power can flow through to the other side.

If you just connec the Ground and the Rx/Tx only the data is connected. The chip in the cable is then powered by the USB side. If you disconnect the cells from the BMS then there is no power flowing from the USB/chip to the BMS.

**Comment by smartpowernorway** [comment]
Hei Louis,

Thanks for the quick response.
I am afraid the cable I am using is not isolated as you mentioned.
Nevertheless, it does not have the red power line on it. It only has
black,green and white which I believe is gnd, tx and Rx (see attached
picture) am I wrong?

I also have a victron USB-RS485  converter or cable which I use with the
Carlo Gavazzi meter. I might try with this one. Is RS485 same as UART
cable??



Med vennlig hilsen,

Ibrahin Chavez
Smart Power AS
+47 47729023

**Reply by Louisvdw** [reply]
If you only have Gnd, Rx, Tx then that is fine. The BMS power should not come from the connection. I would look at the BMS then. If the balance leads and the C- B- are both not connected you should not have power on the BMS.

**Comment by smartpowernorway** [comment]
The test I did... Disconnected the complete connector of balance lines (17
in my case), with B- and P- still connected to battery and inverter and
UART connected to Cerbo. The BMS does not disconnect the power (it does not
open B- to P-)

Med vennlig hilsen,

Ibrahin Chavez
Smart Power AS
+47 47729023

**Reply by Louisvdw** [reply]
If B- and P- is still connected it will power the BMS. You need to disconnect those as well.

**Comment by smartpowernorway** [comment]
But in normal operation B- is connected to battery negative and P- to
inverter. Then if a condition happens the BMS should open the path B- to P-
(open the fets). Am I right?



Med vennlig hilsen,

Ibrahin Chavez
Smart Power AS
+47 47729023

**Reply by Louisvdw** [reply]
Yes, but there is an outstanding issue #17 where I do not yest show when the bank (battery) on offline. Sometimes the request for data from the BMS does not give a reply and the driver currently tries to keep it alive and request again. This keep alive does not show correctly (after a timeout) that the BMS/battery is offline. 
If you disconnect the BMS completely it will be fine though.

**Comment by smartpowernorway** [comment]
Ok, I understand your point now.


Please keep in mind that in "normal operation" I do not want to remove or
disconnect the BMS completely. I just want the BMS to open B- to P- path.

So what is happening now... if some wrong condition happens, the BMS does
not open B- to P-

I understand that if I disconnect them manually then the BMS goes back to
normal.


Do you see my point?

*Med vennlig hilsen,*

*Ibrahin Chavez*
Smart Power AS
+47 47729023

**Reply by Louisvdw** [reply]
Yes that is an issue. But I expect the issue is not with really with the driver, but rather with the BMS.
The driver can just help with the situation. If the driver was to correctly show the GX that the battery is offline, the GX will handle that by not (dis)charging the battery. 
However the BMS is in control of what it does. The driver can only change the GX behaviour by changing the inputs into the ESS parameters. 
The driver does not have any control over your BMS.

**Comment by smartpowernorway** [comment]
You are completely right.

My next test is to power up the Cerbo from an isolated power supply.
Perhaps a second battery I believe this should solve the problem. If it
does, then I might go with an isolated DC to DC converter from 48 to 12 to
power up the Cerbo.


What do you think?

Do you want me to keep you posted?

*Med vennlig hilsen,*

*Ibrahin Chavez*
Smart Power AS
+47 47729023

**Comment by Louisvdw** [comment]
Keep us posted. This could help someone else in future.
I do suspect the issue is with how the Daly handles error conditions. 

From what I have seen with all the BMS that I have worked with on this driver, I will use any in my own system except the Daly. 
They have some wierd logic, problematic parameters (that cannot be customised like all the others) and their protocol implementation makes it hard to get fast updates from the BMS.

**Comment by smartpowernorway** [comment]
Which one would you recommend which is comparable in price to Daly? Or at
least in the same price band
*Med vennlig hilsen,*

*Ibrahin Chavez*
Smart Power AS
+47 47729023

**Comment by Louisvdw** [comment]
You can't go wrong with the JBD. The JKBMS is just a small bit more expensive and have active balancers. Both are similar priced to the Daly Smart.

**Comment by smartpowernorway** [comment]
Thank you very much.


*Med vennlig hilsen,*

*Ibrahin Chavez*
Smart Power AS
+47 47729023

**Comment by smartpowernorway** [comment]
Hei Louis,

I have a project and looking for a smart software guy like you. Off course
you charge me for your work.

I need to connect an energy meter via HAN port to Victron Cerbo GX.

All the protocol info is here:
https://hanporten.se/

It is a serial protocol, I don't think it should be a problem for you.


My other option is to buy a Carlo Gavazzi energy meter which connects via
RS485 (RS485 to USB cable) with the Cerbo GX. But then it would use more
space on my fuse box and the meter costs 350 Euros  with the cable.
https://www.farco.no/tilbehor/tilbehor-invertere/victron-energy-meter-em24-3-phase-max-65a/phase-ethernet
https://www.farco.no/victron/victron-rs485-to-usb-interface-1.8m

Perhaps a good idea is to look into the Carlo Gavazzi protocol:
https://www.gavazzi.no/wp-content/uploads/WM15-kommunikasjonsprotokoll-V1-1-20192510.pdf


I look forward to hearing back from you.

Merry Christmas.

*Med vennlig hilsen,*

*Ibrahin Chavez*
Smart Power AS
+47 47729023

**Comment by smartpowernorway** [comment]
Hi Louis,

I have not received a reply from you so not sure if you are not interested
or missed my email.


As I mentioned, I am willing to pay fairly for your work.

Looking forward to hearing from you.

Best regards

Med vennlig hilsen,

Ibrahin Chavez
Smart Power AS
+47 47729023

**Reply by Louisvdw** [reply]
Hi Ibrahin
Sorry for only replying now. I was away for a few days.
I do not have capacity to start a new project in the next 3 months. You can send me [a PM](https://energytalk.co.za/u/louisvdw/) if that will fit your project timeline, and we can talk.

**Comment by smartpowernorway** [comment]
Hei Louis,

Ok let's for within 3 months.
What do you need to know in advance?
Approximated budget?


*Med vennlig hilsen,*

*Ibrahin Chavez*
Smart Power AS
+47 47729023

**Comment by smartpowernorway** [comment]
Hi Luis,

I agree with waiting for 3 months to start the project.
Do you need anything from me in advance? information, etc?

*Med vennlig hilsen,*

*Ibrahin Chavez*
Smart Power AS
+47 47729023

**Comment by smartpowernorway** [comment]
Hi Louis, I wanted to touch base with you and make sure that you are on board and available to work on my project. You told me 3 months and I am waiting 😊 hoping it can be sooner. Perhaps you could start by taking a look at what I want and from there you can determine how easy or difficult it is. I look forward to hearing back from you.  Med vennlig hilsen, Ibrahin ChavezSmart Power AS+47 47729023  Fra: Smart PowerSendt: 09 January 2022 kl. 11:52Til: Louisvdw/dbus-serialbatteryEmne: Re: [Louisvdw/dbus-serialbattery] Welcome to dbus-serialbattery Discussions! (Discussion #55) Hi Luis, I agree with waiting for 3 months to start the project.Do you need anything from me in advance? information, etc?Med vennlig hilsen, Ibrahin ChavezSmart Power AS+47 47729023  On Tue, Jan 4, 2022 at 8:02 PM Smart Power ***@***.***> wrote:Hei Louis, Ok let's for within 3 months. What do you need to know in advance?Approximated budget? Med vennlig hilsen, Ibrahin ChavezSmart Power AS+47 47729023  On Tue, Jan 4, 2022 at 6:53 AM Louis Van Der Walt ***@***.***> wrote:Hi IbrahinSorry for only replying now. I was away for a few days.I do not have capacity to start a new project in the next 3 months. You can send me a PM if that will fit your project timeline, and we can talk.—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android. You are receiving this because you commented.Message ID: ***@***.***>

**Comment by smartpowernorway** [comment]
Can you please provide me with your personal email? So I don't have to
contact you via Github.
Thank you very much.

*Med vennlig hilsen,*

*Ibrahin Chavez*
Smart Power AS
+47 47729023

**Reply by Louisvdw** [reply]
Hi
Please send me a Personal Message [on this forum](https://energytalk.co.za/u/louisvdw) or [this one](https://diysolarforum.com/members/louisvdw.19047/) and I can send you an email address

Regards

**Comment by smartpowernorway** [comment]
Hi Louis,  I hope you are doing well.  Are you available now to start with our project? I look forward to hearing from you. Med vennlig hilsen, Ibrahin ChavezSmart Power AS+47 47729023  Fra: Louis Van Der WaltSendt: 17 February 2022 kl. 06:44Til: Louisvdw/dbus-serialbatteryKopi: smartpowernorway; CommentEmne: Re: [Louisvdw/dbus-serialbattery] Welcome to dbus-serialbattery Discussions! (Discussion #55) HiPlease send me a Personal Message on this forum or this one and I can send you an email addressRegards—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android. You are receiving this because you commented.Message ID: ***@***.***>

**Comment by SilverFox3333** [comment]
I got started with LiFePO4 batteries in 2019. There weren't many reasonably priced 30kwh choices available in the USA then. So I ordered a rack-mount system from QH Technology in China. It came just in time for the beginning of our Covid lockdowns. It had some poor design flaws which I have gradually corrected over time.

Then this summer I built my own LiFePO4 battery system. The technology available now is so far ahead of what was available to me in 2019 it is quite amazing. I have learned so much, but there is always still room to learn so much more.

My DIY system is 3 racks of (16) LiFePO4 prismatic cells in series, each with its own JK BMS, for a total of an additional 42kwh. These work great!

Back in 2019, it seemed like lithium was lithium, so I built my system in a metal shed detached from my house - in case of fire. Now we come to see that LiFePO4 can melt but won't catch fire. Still, since I had room in my metal shed, I put my new additional batteries in the shed also.

JK wants to communicate with bluetooth. Works great when I go into the shed, but not outside the shed.

So I've been experimenting with getting my batteries (hopefully all of them!) to communicate with my Victron Cerbo GX in the house with various cables.

So many questions arise from this:

For the JK batteries, can dbus-serialbattery handle 3 at once? Can I just parallel the RS485 lines and feed one (isolated) RS485 to USB converter, connected to my Cerbo GX? Or do I need 3 converters and a USB port expander, since the Cerbo only has 2 USB communication port?

For my earlier system from QH Tech, each of the 5 rack modules has its own older Pace BMS. I have successfully gotten them in a daisy chain RS485 to communicate with a laptop PC using one RS485 to USB adapter. On the PC, I can see all 5 Pace BMS's. Unfortunately, these Pace BMS's are so old that they pre-date CAN communication, so there is no Victron link option.

What would be ideal is to have these (5) Pace BMS's also communicate with the Cerbo GX. Then I would have all (8) BMS's visible in Venus GX. 

What are my prospects for accomplishing any of this? Any suggestions or ideas from anyone else who has tried anything like this?

And thanks again for Louis' great program and all the awesome contributions others have made to this project.

**Reply by mr-manuel** [reply]
> For the JK batteries, can dbus-serialbattery handle 3 at once? Can I just parallel the RS485 lines and feed one (isolated) RS485 to USB converter, connected to my Cerbo GX? Or do I need 3 converters and a USB port expander, since the Cerbo only has 2 USB communication port?

* https://louisvdw.github.io/dbus-serialbattery/faq/#how-many-usb-to-serial-adapters-can-i-connect
* At the state of today, you cannot parallel them. One serial converter per battery is needed
* https://louisvdw.github.io/dbus-serialbattery/faq/#what-can-i-do-if-the-bms-communication-is-unstable make sure to buy a cheap USB hub

> What would be ideal is to have these (5) Pace BMS's also communicate with the Cerbo GX. Then I would have all (8) BMS's visible in Venus GX.
>
> What are my prospects for accomplishing any of this? Any suggestions or ideas from anyone else who has tried anything like this?

If you get data from them you can use Node-RED (which is integrated in Venus OS Large, just google for it) and [dbus-mqtt-battery](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery) for each battery.

Probably the Cerbo GX can't handle that much batteries since it has not that much CPU power as a Raspberry Pi. I would reccomend you to get a Raspberry Pi 4, install Venus OS large, connect all batteries there, [aggregate all batteries](https://louisvdw.github.io/dbus-serialbattery/faq/#how-to-aggregate-multiple-batteries) with a battery aggregator, install dbus-mqtt-battery on Cerbo GX and feed it with the data from the aggregator.

**Comment by serveroutlet** [comment]
This is a very powerful project, thank you.

I'm just starting to mess with it and hopefully someone can assist, but my issue appears to be the SOC for the JBD bms units. The current latest version that is available for download to a flash drive is showing incorrect SOC values, about double what the xiaoxing app is showing.
But everything else is working properly including custom battery names and temperature probe names.

I also saw the SSH latest release, this shows the correct SOC, but it no longer supports custom battery names or custom names for the probe.s

Does anyone have some leads on a version that shows the correct SOC and and also supports custom battery names and temperature probe names?

**Comment by serveroutlet** [comment]
Ok I will try:
https://github.com/Louisvdw/dbus-serialbattery/tree/v1.0.20231125beta

as it has the custom names in config.ini

cheers.

Sent with [Proton Mail](https://proton.me/) secure email.

**Comment by serveroutlet** [comment]
Yes so this version has an issue it is doubling the JBD stated SOC capacity when compared to the xiaoxing app.
https://github.com/Louisvdw/dbus-serialbattery/tree/v1.0.20231125beta

This version reports the correct SOC (matches the xiaoxing app) but does not support custom names the options are actually missing form the ini file completely:
https://github.com/Louisvdw/dbus-serialbattery/tree/v1.0.20240102beta

Thank you for your time.

Thanks

Sent with [Proton Mail](https://proton.me/) secure email.

**Reply by mr-manuel** [reply]
Stop spamming please. You post the same question in 1.000 places at the same time. This is helping nobody and generates a lot of work only because you can't wait for an answer.

https://github.com/Louisvdw/dbus-serialbattery/discussions/931

**Comment by Maxmat99** [comment]
hello ! 
system is this:
4x48/5000 multis 
victron mppt
1x fronius 3000 pv inverter
1x sunny boy 4000 pv inverter
pi 3b running venus 
6x 12s 3.7v batteries with grrrhhh dly bms will change in the future 
dbs serial (amazing lol)
kindred battery aggregator....
running a campsite in the uk 
off grid......

brilliant allworks well with 1.0.20230531 nice and stable 
if i go any higher with dbs version it doesnt pick up all 6 bms ! really annoying as id like to use a lot of the features! any ideas ?
another thing i would like some help with is i have the sunny boy,, with serial adpter feeding an old webbox, this is not picked up by venus. has any one done any code to help pick up modbus signals only just for energy meter!

i am open to anyone who wants to help remotely troubleshoot/tweak etc ....
however this would have to be not on weekends as the site is relying on power everything!
great jobs peeps!!
keep it going!!
i have tailscale installed etc..

**Reply by mr-manuel** [reply]
Best would be to read the pinned discussions ;)

![grafik](https://github.com/Louisvdw/dbus-serialbattery/assets/12592765/aaeab1aa-d8c1-466c-8a11-30c9e55b6fb4)

https://github.com/Louisvdw/dbus-serialbattery/discussions/856

**Comment by jakub199** [comment]
Hi, please also woks with Multiplus II GX (Venus included in inverter) ?

**Reply by Louisvdw** [reply]
Hi
Yes it works the same with the internal GX devices as with the seperate GX (like CerboGX)

**Comment by NickRoberts174** [comment]
Hi all,

I was wondering if there's any chance of having some more installation info on the steps needed to get single / parallel BMS's working over CANbus? 

I have read the documentation, changed the CAN address in the default settings and still no closer to understanding why my BMS's aren't showing up. 

Hate to have to ask for help but there so little information/discussion on this. To give an example of what I've been trying to achieve, my set up is like this.

Cerbo GX 
Daly Smart BMS (x2 off)
WNT board (x2 off)
CANbus cable to connect Daly-Victron (type B from memory) 
RX and TX messages are being passed and can be seen on candump can0  

Tried with and without line resistors (even made an adapter to convert the line resistor work with the pinout of the WNT board)
Tried several combinations on the dip switches with no effect
Tried hooking both up with their own serial USB converters to the RS485 jacks, this only showed one battery but the values were nonsense or would flick between Battery 1 and Battery 2 and then disconnect after a short while

Just some guidance in the documentation how to set this up would be super helpful.  

![image](https://github.com/user-attachments/assets/4cbed091-b7d7-4947-8ab3-03054df74ed2)

![image](https://github.com/user-attachments/assets/082b4041-6e34-4712-8e7e-3eacfdce60e1)
